crates/rethnet_evm/benches/util.rs

```diff
@@ -48,6 +49,9 @@ impl RethnetStates {
                 }
                 state.checkpoint().unwrap();
             }
+            for _ in 0..number_of_snapshots {
```

Comment:
Make the same snapshots over and over will not result in multiple snapshots, as the state root will be the same. You'd need to employ the same approach as is done for checkpoints: every X changes, create a snapshot

Reply:
see how i'm now varying the balance in the new function `prep_snapshots()`